src/neutron/universaladdr.h

```diff
@@ -0,0 +1,57 @@
+//
+// Created by CodeFace on 2019-08-27.
+//
+
+#ifndef QTUM_NEUTRON_UNIVERSALADDR_H
+#define QTUM_NEUTRON_UNIVERSALADDR_H
+
+#include <string>
+#include <script/standard.h>
+
+class UniversalAddress {
+public:
+    enum Version: uint16_t {
+        UNKNOWN         = 0x0000,
+        P2PKH           = 0x0001,   // pay to pubkey hash
+        P2SH            = 0x0002,   // pay to script hash
+        P2WPKH          = 0x0003,   // pay to witness pubkey hash
+        P2WSH           = 0x0004,   // pay to witness script hash
+        NONSTANDARD     = 0x0005,   // non-standard
+        LEVM            = 0x4001,   // legacy EVM
+        NEVM            = 0x8002,   // Neutron EVM
+        NX86            = 0x8003,   // Neutron X86
+        NTVM            = 0x8004    // Neutron Test VM
+    };
+
+    UniversalAddress();
+    UniversalAddress(Version version, std::vector<unsigned char> data);
+    UniversalAddress(Version version, const unsigned char *data, size_t size);
+    UniversalAddress(const UniversalAddress &another);
+
+    ~UniversalAddress() {
+    }
+
+    UniversalAddress& operator = (const UniversalAddress &another);
+    bool operator == (const UniversalAddress &another);
+
+    inline Version version() const                          { return m_version; }
+    inline const std::vector<unsigned char>& data() const   { return m_data; };
+
+    inline void setVersion(Version version)         { m_version = version; }
+    void setData(const std::vector<unsigned char> &data);
+    void setData(const unsigned char *data, size_t size);
+    int setHex(const char* psz);
+    int setHex(const std::string &str);
+
+    std::string getHex();
+
+private:
+    Version m_version;
```

Comment:
Where is the Version class defined? I don't see it here

Reply:
it is inside the class UniviersalAddress, see https://github.com/qtumproject/qtum-neutron/blob/bb1f2981d685a651c27982505edcfaaf24f65f91/src/neutron/universaladdr.h#L13

Reply:
Oh, doh! I was thought this was a class with a constructor or something.  Code looks good so far, just figure out some tests